### PR TITLE
feat: add market creation task

### PR DESCRIPTION
**Pull request by ramenforbreakfast**
This market creation task will:
- Generate the market address ahead of time for preparation purposes
- Deploy contracts peripheral to the `PrePOMarket` and perform contract alignments along with basic default configuration (calling the helpers defined in PR #865)
- Register the contract ahead of time in the OZ database (for reasons explained in the comments)
- Submit a batch proposal to create a market, and set the mint and redeem hooks to the ones that were just deployed.

### CODE REVIEW

File: apps/smart-contracts/core/tasks/markets.ts [hunk 1]
```diff
@@ -92,3 +98,203 @@ const prepareMarketPeriphery = async (
...
+  )
+  .addOptionalParam(
+    'floorValuation',
+    'floor valuation of asset in hundred millions of $ to 2 decimal places e.g. $123 million = 123',
+    5000,
+    types.int
+  )
+  .addOptionalParam(
+    'ceilingValuation',
+    'ceiling valuation of asset in hundred millions of $ to 2 decimal places e.g. $123 million = 123',
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/tasks/markets.ts:125]:
```suggestion
    "ceiling valuation of asset in millions of $ e.g. '123' = $123M",
```

Comment (XavierEkkel) [reply apps/smart-contracts/core/tasks/markets.ts:125]: same for above

File: apps/smart-contracts/core/tasks/markets.ts [hunk 2]
```diff
@@ -92,3 +98,203 @@ const prepareMarketPeriphery = async (
...
+  .addParam('symbolSuffix', 'suffix of market token symbol e.g. preSTRIPE_5000-10000_30SEP23')
+  .addOptionalParam(
+    'floorPrice',
+    "floor price for Long token position, ethers string e.g. '1' = 1 ether.",
+    '0.5',
+    types.string
+  )
+  .addOptionalParam(
+    'ceilingPrice',
+    "ceiling price for Long token position, ethers string e.g. '1' = 1 ether.",
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/tasks/markets.ts:113]:
```suggestion
    "floor payout in ether units for Long token position e.g. '0.5' = 0.5 ether = 50%",
    '0.5',
    types.string
  )
  .addOptionalParam(
    'ceilingPrice',
    "ceiling payout in ether units for Long token e.g. '1' = 1 ether = 100%",
```

**Review by XavierEkkel — APPROVED** [review]
LGTM after nits

File: apps/smart-contracts/core/tasks/markets.ts [hunk 3]
```diff
@@ -92,3 +98,203 @@ const prepareMarketPeriphery = async (
...
   )
 }
+
+task('create-market', 'create PrePOMarket from PrePOMarketFactory')
+  .addParam('nameSuffix', 'suffix of market token name e.g. preSTRIPE 5000-10000 30-September 2023')
+  .addParam('symbolSuffix', 'suffix of market token symbol e.g. preSTRIPE_5000-10000_30SEP23')
+  .addOptionalParam(
+    'floorPrice',
+    "floor price for Long token position, ethers string e.g. '1' = 1 ether.",
+    '0.5',
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/tasks/markets.ts:108]: why are all these param values prefilled?

Comment (ramenforbreakfast) [reply apps/smart-contracts/core/tasks/markets.ts:108]: Yeah this was more for convenience during testing, I think for safety purposes, I will remove them and make them required parameters to define.

**Pull request merged**